File: python/palindrome_test.py
import unittest
from palindrome import isPalindrome

class Palindrome(unittest.TestCase):
    def setUp(self):
        self.testcase1 = ''
        self.testcase2 = 'a'
        self.testcase3 = 'aa'
        self.testcase4 = 'aba'
        self.testcase5 = 'abc'
        self.testcase6 = 'tacocat'
        self.testcase7 = 'racecar'
        self.testcase8 = 'notpalindrome'
    
    def tearDown(self):
        """Executed after each test"""
        pass

    def test_is_empty_sting_palindrome(self):
        is_palindrome = isPalindrome(self.testcase1)

        self.assertEqual(is_palindrome, True)
    
    def test_is_a_palindrome(self):
        is_palindrome = isPalindrome(self.testcase2)

        self.assertEqual(is_palindrome, True)
    
    def test_is_aa_palindrome(self):
        is_palindrome = isPalindrome(self.testcase3)

        self.assertEqual(is_palindrome, True)
    
    def test_is_aba_palindrome(self):
        is_palindrome = isPalindrome(self.testcase4)

        self.assertEqual(is_palindrome, True)
    
    def test_is_abc_palindrome(self):
        is_palindrome = isPalindrome(self.testcase5)

        self.assertEqual(is_palindrome, False)
    
    def test_is_tacocat_palindrome(self):
        is_palindrome = isPalindrome(self.testcase6)

        self.assertEqual(is_palindrome, True)
    
    def test_is_racecar_palindrome(self):
        is_palindrome = isPalindrome(self.testcase7)

        self.assertEqual(is_palindrome, True)
    
    def test_is_notpalindrome_palindrome(self):
        is_palindrome = isPalindrome(self.testcase8)

        self.assertEqual(is_palindrome, False)

if __name__ == '__main__':
    unittest.main()

File: typescript/tests/palindrome.test.ts
import { isPalindrome } from '../src/palindrome'

import { expect } from 'chai'

describe('isPalindrome tests', () => {
  describe('when given a string of length 0', () => {
    it('returns true', () => {
      expect(isPalindrome('')).to.be.true;
    })
  })

  describe('when given a string of length 1', () => {
    it('returns true', () => {
      expect(isPalindrome('a')).to.be.true;
    })
  })

  describe('when given string "aa"', () => {
    it('returns true', () => {
      expect(isPalindrome('aa')).to.be.true;
    })
  })

  describe('when given string "aba"', () => {
    it('returns true', () => {
      expect(isPalindrome('aba')).to.be.true;
    })
  })

  describe('when given string "abc"', () => {
    it('returns false', () => {
      expect(isPalindrome('abc')).to.be.false;
    })
  })

  describe('when given string "tacocat"', () => {
    it('returns true', () => {
      expect(isPalindrome('tacocat')).to.be.true;
    })
  })

  describe('when given string "racecar"', () => {
    it('returns true', () => {
      expect(isPalindrome('racecar')).to.be.true;
    })
  })

  describe('when given string "notpalindrome"', () => {
    it('returns false', () => {
      expect(isPalindrome('notpalindrome')).to.be.false;
    })
  })
})
